daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/transaction/PartialTransaction.scala

```diff
@@ -54,18 +54,25 @@ private[lf] object PartialTransaction {
   }
 
   private[PartialTransaction] object NoneSeededTransactionRootContext extends RootContextInfo {
-    val childSeed: Any => Nothing = { _ =>
+    val actionChildSeed: Any => Nothing = { _ =>
       throw new IllegalStateException(s"the machine is not configure to create transaction")
     }
   }
 
   /** Contexts of the transaction graph builder, which we use to record
     * the sub-transaction structure due to 'exercises' statements.
     */
-  final case class Context(info: ContextInfo, children: BackStack[NodeId], nextChildIdx: Int) {
-    def addChild(child: NodeId): Context = Context(info, children :+ child, nextChildIdx + 1)
+  final case class Context(
+      info: ContextInfo,
+      children: BackStack[NodeId],
+      nextActionChildIdx: Int,
+  ) {
+    def addActionChild(child: NodeId): Context =
+      Context(info, children :+ child, nextActionChildIdx + 1)
+    def addRollbackChild(child: NodeId, nextActionChildIdx: Int): Context =
+      Context(info, children :+ child, nextActionChildIdx)
```

Comment:
We decided some time ago,  that the rollback node will not interact with the contract ID scheme, the contract ID scheme will consider only Action Node. This was the purpose of #8808.
So since the rollback node is not an action we do not increment the `nextActionChildIdx`

Reply:
I think I’m still slightly confused. The rollback node itself should not affect the CID scheme but children of that node which we’re adding here do don’t they? If they don’t, I would expect CID collisions between contracts created in rollback nodes and those created outside which we don’t want.

Reply:
Oh I see what you’re saying. I got slightly confused by the naming. `addRollbackChild` does not add a child to a rollback node. It adds a rollback node as the child of whatever is the parent. makes sense :+1: